jmclient/jmclient/wallet.py

```diff
@@ -181,18 +208,52 @@ def add_utxo(self, txid, index, path, value, mixdepth):
 
         self._utxo[mixdepth][(txid, index)] = (path, value)
 
+    def is_disabled(self, txid, index):
+        if self._utxo_meta is None:
+            return False
+        if (txid, index) not in self._utxo_meta.keys():
+            return False
+        if b'disabled' not in self._utxo_meta[(txid, index)]:
+            return False
+        if not self._utxo_meta[(txid, index)][b'disabled']:
+            return False
+        return True
+
+    def disable_utxo(self, txid, index, disable=True):
+        assert isinstance(txid, bytes)
+        assert len(txid) == self.TXID_LEN
+        assert isinstance(index, numbers.Integral)
+
+        if b'disabled' not in self._utxo_meta[(txid, index)]:
+            self._utxo_meta[(txid, index)] = {}
+        self._utxo_meta[(txid, index)][b'disabled'] = disable
+
+    def enable_utxo(self, txid, index):
+        self.disable_utxo(txid, index, disable=False)
+
     def select_utxos(self, mixdepth, amount, utxo_filter=(), select_fn=None):
         assert isinstance(mixdepth, numbers.Integral)
         utxos = self._utxo[mixdepth]
+        # do not select anything in the filter
         available = [{'utxo': utxo, 'value': val}
-            for utxo, (addr, val) in utxos.items() if utxo not in utxo_filter]
+        for utxo, (addr, val) in utxos.items() if utxo not in utxo_filter]
+        # do not select anything disabled
+        available = [u for u in available if not self.is_disabled(*u['utxo'])]
```

Comment:
couldn't this be combined with the list comprehension above? Refactoring this into a `for` loop would make sense in that case.

Reply:
Probably. I'm going to leave this alone for now if that's OK with you. It's understandable what it does, right.

Reply:
sure, the code works either way.